halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/oraclebmcs/OracleBMCSAccountValidator.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2017 Oracle America, Inc.
+ *
+ * The contents of this file are subject to the Apache License Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * If a copy of the Apache License Version 2.0 was not distributed with this file,
+ * You can obtain one at https://www.apache.org/licenses/LICENSE-2.0.html
+ */
+
+package com.netflix.spinnaker.halyard.config.validate.v1.providers.oraclebmcs;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.oraclebmcs.OracleBMCSAccount;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import com.netflix.spinnaker.halyard.core.problem.v1.Problem.Severity;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OracleBMCSAccountValidator extends Validator<OracleBMCSAccount> {
+  @Override
+  public void validate(ConfigProblemSetBuilder psBuilder, OracleBMCSAccount account) {
+
+    notNullOrEmpty(account.getCompartmentId(), "compartment id", psBuilder);
```

Comment:
Since these are all required, in the `account add` subcommand I would add the `@Parameter` property `required = true`.